copilot-instructions.md

```diff
@@ -0,0 +1,408 @@
+- Event handlers: prefix with `handle` or `on`: `handleClick`, `onSubmit`
+
+Components (React):
+- Use PascalCase: `SitecoreComponent`, `PageLayout`, `ContentBlock`
+- File names match component names: `SitecoreComponent.tsx`
+
+Constants:
+- Use UPPER_SNAKE_CASE: `API_ENDPOINT`, `DEFAULT_TIMEOUT`, `MAX_RETRIES`
+- Export at module level when shared
+
+Directories:
+- Use kebab-case: `src/components`, `src/api-clients`, `src/sitecore-utils`
+- Organize by feature when appropriate: `src/content-management/`
+
+Types and Interfaces:
+- Use PascalCase with descriptive names: `ContentItem`, `LayoutProps`, `SitecoreConfig`
+- Prefix interfaces with `I` only when needed for disambiguation
+
+### Code Layout and Organization
+
+Directory Structure:
+```
+src/
+  components/          # UI components (React)
+  utils/               # Helper functions and utilities
+  api/                 # Sitecore integrations and API clients
+  lib/                 # Third-party library configurations
+  types/               # TypeScript type definitions
+  hooks/               # Custom React hooks
+  styles/              # Styling files
+```
+
+File Organization:
+- Group related functionality in feature directories
+- Keep components co-located with their styles and tests
+- Export public APIs through index.ts files
+
+### Error Handling
+
+API Calls:
+- Always wrap in try/catch blocks
+- Throw custom errors with context: `SitecoreFetchError`, `ConfigurationError`
+- Handle edge cases with guard clauses
+
+### Security
+
+Input Validation:
+- Sanitize user inputs before processing
+- Validate data at application boundaries
+- Use type guards for runtime type checking
+- Escape content when rendering to prevent XSS
+
+### Performance
+
+Optimization Patterns:
+- Memoize components with React.memo when appropriate
+- Lazy-load non-critical modules: `const Component = lazy(() => import('./Component'))`
+- Use useCallback and useMemo for expensive operations
+
+TypeScript:
+- Enable strict mode in tsconfig.json
+- Prefer type assertions over any: `value as ContentItem`
+- Use discriminated unions for complex state management
+
+### Documentation
+
+JSDoc Comments:
+- All new functions, interfaces, classes must have JSDoc style comments
+- Include @param tags for all parameters with types and descriptions
+- Include @returns tag for return values with type and description
+- Use descriptive comments that explain the purpose and behavior
+- Follow existing Content SDK JSDoc patterns
+
+## Sitecore SDK Rules
+
+### XM Cloud Integration
+
+Configuration:
+- Always use environment variables for API endpoints and keys
+- Never hardcode API keys in source code
+- Use `.env.example` files to document required environment variables
+- Prefer `sitecore.config.ts` for centralized configuration
+
+```typescript
+// sitecore.config.ts
+export const sitecoreConfig = {
+  sitecoreApiHost: process.env.SITECORE_API_HOST || '',
+  sitecoreApiKey: process.env.SITECORE_API_KEY || '',
```

Comment:
Would be better to provide a default example using client/server context id:
Also our interface is a bit different:
```ts
api: {
   edge: { ... }
}
```